### ISSUE TITLE
warning when using masked arrays for ang2pix

### ISSUE THREAD

**Issue by segasai** [healpy/healpy#645]
Hi, 

I don't quite know the internals of how healpy behaves with masked arrays, but currently
the following example produces a warning which I find misleading
```
In [40]: healpy.ang2pix(1024,np.ma.array([1,2.],mask=False,fill_value=1e20),np.arange(2))
/home/koposov/pyenv3/lib/python3.6/site-packages/healpy/pixelfunc.py:475: RuntimeWarning: invalid value encountered in _ang2pix_ring
  return pixlib._ang2pix_ring(nside, theta, phi)
Out[40]: 
masked_array(data=[2893824, 8907404],
             mask=False,
       fill_value=1e+20)
 ```

It is not a big deal, but I've just  spend an trying to chase the origin of the warning thinking that some of my inputs to healpy are corrupted.

**Comment by zonca** [comment]
unfortunately it's coming from the bindings with C++, somewhere around https://github.com/healpy/healpy/blob/master/healpy/src/_healpy_pixel_lib.cc#L42, so I have no idea how to fix this.

And I wouldn't want to blindly filter out all `RuntimeWarning`, it could hide a case where there is actually a proper error.

Any suggestions?

**Comment by segasai** [comment]
Hmm. 
I am not sure I have suggestion. First I don't really understand why the c++ code is called on invalid data in the first place (because the masked array does not have any values masked at all). My current hypothesis is that the c++ interface is callled on the fill value... But I don't understand enough ufunc machinery to test this.

**Comment by mreineck** [comment]
Is it possible that everything goes basically well, but that at the end of the function some checking code detects a type mismatch between the masked array's data (integer) and the fill value (float)? I'm not sure how the fill value gets propagated to the output array, but it certainly makes no sense there.

**Comment by zonca** [comment]
no, the first array is already a float (one of the inputs as a dot), I also tested having both arrays float64:

     healpy.ang2pix(1024,np.ma.array([1,2.],mask=False,fill_value=1e20, dtype=np.float64), np.arange(2, dtype=np.float64))

but gives the same error, so it must be something related to the masked array.
